### PR TITLE
bpo-25910: Fixes redirection from http to https

### PR DESCRIPTION
**Pull request by CuriousLearner**
This PR extracts changes from https://github.com/python/cpython/pull/1933

Currently focuses mostly on fixing link redirection from http to https.

cc @vstinner 

https://bugs.python.org/issue25910

### CODE REVIEW

File: Doc/faq/gui.rst [hunk 1]
```diff
@@ -43,11 +43,11 @@ number of platforms, with Windows, Mac OS X, GTK, X11, all listed as
 current stable targets.  Language bindings are available for a number
 of languages including Python, Perl, Ruby, etc.
 
-wxPython (http://www.wxpython.org) is the Python binding for
-wxwidgets.  While it often lags slightly behind the official wxWidgets
-releases, it also offers a number of features via pure Python
-extensions that are not available in other language bindings.  There
-is an active wxPython user and developer community.
+wxPython (https://www.wxpython.org) is the Python binding for wxwidgets.
```
Comment (vstinner) [inline Doc/faq/gui.rst]: Would you mind to create a link?

```
`wxPython <url>`_
```

File: Doc/faq/gui.rst [hunk 2]
```diff
@@ -43,11 +43,11 @@ number of platforms, with Windows, Mac OS X, GTK, X11, all listed as
...
+While it often lags slightly behind the official wxWidgets releases, it also
```
Comment (vstinner) [inline Doc/faq/gui.rst]: @vadmium asked you to not reformat links like this, please restrict this PR to HTTPS.

File: Doc/faq/windows.rst [hunk 3]
```diff
@@ -170,8 +170,8 @@ offender.
 How do I make an executable from a Python script?
 -------------------------------------------------
 
-See http://cx-freeze.sourceforge.net/ for a distutils extension that allows you
-to create console and GUI executables from Python code.
+See https://anthony-tuininga.github.io/cx_Freeze/ for a distutils extension
```
Comment (vstinner) [inline Doc/faq/windows.rst]: Please create a link. You may be able to omit the url here, since it's written in a previous section (check the HTML output, I'm not sure).

```
`cx_Freeze`_
```

File: Doc/library/hashlib.rst [hunk 4]
```diff
@@ -694,7 +694,7 @@ implementation, extension code, and this documentation:
 
    You should have received a copy of the CC0 Public Domain Dedication along
    with this software. If not, see
-   http://creativecommons.org/publicdomain/zero/1.0/.
+   https://creativecommons.org/publicdomain/zero/1.0/.
```
Comment (vstinner) [inline Doc/library/hashlib.rst:697]: This text is  a license. Is it ok to modify a license?

File: Doc/using/windows.rst [hunk 5]
```diff
@@ -871,11 +871,11 @@ shipped with PyWin32.  It is an embeddable IDE with a built-in debugger.
...
-`cx_Freeze <http://cx-freeze.sourceforge.net/>`_ is a :mod:`distutils`
-extension (see :ref:`extending-distutils`) which wraps Python scripts into
-executable Windows programs (:file:`{*}.exe` files).  When you have done this,
-you can distribute your application without requiring your users to install
-Python.
+`cx_Freeze <https://anthony-tuininga.github.io/cx_Freeze/>`_ is a
+:mod:`distutils` extension (see :ref:`extending-distutils`) which wraps
+Python scripts into executable Windows programs (:file:`{*}.exe` files).
+When you have done this, you can distribute your application without
+requiring your users to install Python.
```
Comment (vstinner) [inline Doc/using/windows.rst]: Again, please revert this change, only replace the HTTP link.

File: Doc/whatsnew/2.7.rst [hunk 6]
```diff
@@ -1832,8 +1832,8 @@ https://pypi.python.org/pypi/unittest2.
 
 When used from the command line, the module can automatically discover
 tests.  It's not as fancy as `py.test <http://pytest.org>`__ or
-`nose <http://code.google.com/p/python-nose/>`__, but provides a simple way
-to run tests kept within a set of package directories.  For example,
+`nose <https://code.google.com/archive/p/python-nose>`__, but provides a
```
Comment (vstinner) [inline Doc/whatsnew/2.7.rst]: This link seems undated. The new home is more https://nose.readthedocs.io/ no?

File: Doc/whatsnew/3.6.rst [hunk 7]
```diff
@@ -238,8 +238,8 @@ and the ``__annotations__`` attribute.
       and Guido van Rossum. Implemented by Ivan Levkivskyi.
 
    Tools that use or will use the new syntax:
-   `mypy <http://github.com/python/mypy>`_,
-   `pytype <http://github.com/google/pytype>`_, PyCharm, etc.
+   `mypy <https://github.com/python/mypy>`_,
```
Comment (vstinner) [inline Doc/whatsnew/3.6.rst]: The home is more http://www.mypy-lang.org/ no?

**Review by vstinner — CHANGES_REQUESTED** [review]

File: Doc/faq/gui.rst (hunk 2 above)
Comment (CuriousLearner) [reply Doc/faq/gui.rst]: Apologies for that. I just extracted the same old changes from the earlier patch. I'll fix this.

File: Doc/library/hashlib.rst (hunk 4 above)
Comment (CuriousLearner) [reply Doc/library/hashlib.rst:697]: I'm not sure about that. Should I ask this on the bug?

File: Doc/whatsnew/3.6.rst (hunk 7 above)
Comment (CuriousLearner) [reply Doc/whatsnew/3.6.rst]: Updated.

File: Doc/whatsnew/2.7.rst (hunk 6 above)
Comment (CuriousLearner) [reply Doc/whatsnew/2.7.rst]: Seems like you're right about it. The `make linkcheck` command showed me this as a 301. So, I updated it to point to this link.

**Comment by CuriousLearner** [comment]
I have made the requested changes; please review again

File: Doc/library/hashlib.rst (hunk 4 above)
Comment (vstinner) [reply Doc/library/hashlib.rst:697]: I think it's ok. I take the responsability for this change ;-)

**Review by vstinner — APPROVED** [review]
LGTM.

**Comment by vstinner** [comment]
Thanks @CuriousLearner, I merged your PR.

File: Doc/library/hashlib.rst (hunk 4 above)
Comment (CuriousLearner) [reply Doc/library/hashlib.rst:697]: Alright thanks :)

**Comment by CuriousLearner** [comment]
Thanks @vstinner.
